model/ftn/ww3_uprstr.ftn

```diff
@@ -934,15 +1404,398 @@
+!
+!  6. Error messages :
+!
+!  7. Remarks :
+!
+!  8. Structure :
+!
+!  9. Switches :
+!
+!     !/T
+!
+! 10. Source code :
+!
+!/
+      SUBROUTINE SWH_RSRT_1pw (VA1p, WS, WD, ISEA1p, HSIG1p, HSIGwp, HSIGsp, VAMAPWS )  
+!
+         USE W3GDATMD, ONLY: TH
+         USE W3ODATMD, ONLY: WSMULT !same wind sea cut-off factor for sea/swell outputs
+!         
+         REAL, INTENT(OUT) :: HSIG1p, HSIGwp, HSIGsp
+         INTEGER, INTENT(IN) :: ISEA1p
+         REAL, INTENT(IN) :: WS, WD
+         REAL, DIMENSION(:), INTENT(IN)  :: VA1p
+         INTEGER, DIMENSION(:), INTENT(OUT) :: VAMAPWS ! Wind-sea id for spectral bins
+         REAL :: RELWS, ETOTw, ETOTs, EwI, EsI
+         CHARACTER(25),PARAMETER :: myname='SWH_RSRT_1pw'
+!
+!/T         WRITE (NDSO,*) trim(myname), ' starts'         
+         HSIG1p = 0.
+         HSIGwp = 0.
+         HSIGsp = 0.
+         DEPTH  = MAX ( DMIN , -ZB(ISEA1p) )
+         ETOT   = 0.
+         ETOTw  = 0.
+         ETOTs  = 0.
+!
+         DO IK=1, NK
+            CALL WAVNU1 ( SIG(IK), DEPTH, WN, CG )
+            E1I    = 0.
+            EwI    = 0.
+            EsI    = 0.
+            DO ITH=1, NTH
+               ! Relative wind-sea calc assumes input with in direction toward
+               ! i.e. same as for the wave spectrum
+               RELWS = WSMULT * WS * MAX(0.0, COS(WD - TH(ITH)))
+               E1I    = E1I + VA1p(ITH+(IK-1)*NTH) * SIG(IK) / CG
+               IF ( RELWS > (SIG(IK)/WN) ) THEN
+                   EwI = EwI + VA1p(ITH+(IK-1)*NTH) * SIG(IK) / CG
+                   VAMAPWS(ITH+(IK-1)*NTH) = 1
+               ELSE
+                   EsI = EsI + VA1p(ITH+(IK-1)*NTH) * SIG(IK) / CG
+                   VAMAPWS(ITH+(IK-1)*NTH) = 0
+               END IF
+            END DO
+            ETOT   = ETOT + E1I*DSIP(IK)
+            ETOTw  = ETOTw + EwI*DSIP(IK)
+            ETOTs  = ETOTs + EsI*DSIP(IK)
+         END DO
+!
+         HSIG1p = 4. * SQRT ( ETOT * DTH )
+         HSIGwp = 4. * SQRT ( ETOTw * DTH )
+         HSIGsp = 4. * SQRT ( ETOTs * DTH )
+!
+!/T       WRITE (NDSO,*) trim(myname), ' ends'         
+      END SUBROUTINE SWH_RSRT_1pw
+!/
+!/ ------------------------------------------------------------------- /
+!/ 
+!/                  +-----------------------------------+
+!/                  | WAVEWATCH III           NOAA/NCEP |
+!/                  |         Andy Saulter              |
+!/                  |                        FORTRAN 90 |
+!/                  | Last update :         16-Oct-2018 |
+!/                  +-----------------------------------+
+!/
+!/    16-Oct-2018 : Original Code                       ( version 6.07 )
+!/
+!/    Copyright 2010 National Weather Service (NWS),
+!/    National Oceanic and Atmospheric Administration.  All rights
+!/    reserved.  WAVEWATCH III is a trademark of the NWS.
+!/    No unauthorized use without permission.
+!/
+!  1. Purpose :
+!     Calculate the significant wave height for total, wind sea and
```

Comment:
Update purpose to reflect actual purpose!

Reply:
Done